raspi-config

```diff
@@ -56,79 +66,109 @@ else
-  if grep -q "^Revision\s*:\s*00[0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo; then
-    return 0
-  elif grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]0[0-36][0-9a-fA-F]$" /proc/cpuinfo ; then
-    return 0
-  else
-    return 1
+  if [ -z "$IS_PIONE" ]; then
+    if grep -q "^Revision\s*:\s*00[0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo; then
+      IS_PIONE=0
+    elif grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]0[0-36][0-9a-fA-F]$" /proc/cpuinfo ; then
+      IS_PIONE=0
+    else
+      IS_PIONE=1
+    fi
   fi
+  return $IS_PIONE
 }
 
 is_pitwo() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]04[0-9a-fA-F]$" /proc/cpuinfo
-  return $?
+  if [ -z "$IS_PITWO" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]04[0-9a-fA-F]$" /proc/cpuinfo
+    IS_PITWO=$?
+  fi
+  return $IS_PITWO
 }
 
 is_pizero() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]0[9cC][0-9a-fA-F]$" /proc/cpuinfo
-  return $?
+  if [ -z "$IS_PIZERO" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]0[9cC][0-9a-fA-F]$" /proc/cpuinfo
+    IS_PIZERO=$?
+  fi
+  return $IS_PIZERO
 }
 
-# ...while tests for Pi 3 and 4 just test processor type, so will also find CM3, CM4, Zero 2 etc.
+is_cmfive() {
+  if [ -z "$IS_CMFIVE" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]1[8aA][0-9a-fA-F]$" /proc/cpuinfo
+    IS_CMFIVE=$?
+  fi
+  return $IS_CMFIVE
+}
+
+# ...while tests for Pi 3, 4 and 5 just test processor type, so will also find CM3, CM4, Zero 2 etc.
 
 is_pithree() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]2[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
-  return $?
+  if [ -z "$IS_PITHREE" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]2[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
+    IS_PITHREE=$?
+  fi
+  return $IS_PITHREE
 }
 
 is_pifour() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]3[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
-  return $?
+  if [ -z "$IS_PIFOUR" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]3[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
+    IS_PIFOUR=$?
+  fi
+  return $IS_PIFOUR
 }
 
 is_pifive() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]4[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
-  return $?
-}
-
-is_cmfive() {
-  grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]1[8aA][0-9a-fA-F]$" /proc/cpuinfo
-  return $?
+  if [ -z "$IS_PIFIVE" ]; then
+    grep -q "^Revision\s*:\s*[ 123][0-9a-fA-F][0-9a-fA-F]4[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]$" /proc/cpuinfo
+    IS_PIFIVE=$?
+  fi
+  return $IS_PIFIVE
 }
 
 get_pi_type() {
-  if is_pione; then
-    echo 1
-  elif is_pitwo; then
-    echo 2
-  elif is_pithree; then
-    echo 3
-  elif is_pifour; then
-    echo 4
-  elif is_pifive; then
-    echo 5
-  elif is_pizero; then
-    echo 0
-  else
-    echo -1
+  if [ -z "$PI_TYPE" ]; then
```

Comment:
Shouldn't this test GET_PI_TYPE, not PI_TYPE? You are testing one variable and setting another.

Reply:
Yes! :man_facepalming:  I guess I renamed the variable, but forgot to rename all occurrences.